XR Hete klimaatkalender - webshop Woo Commerce

tomootes opened the pull request:
This Pull Request contains all the styling and plugins needed for setting up the webshop for the XR hete klimaat kalender. I added a new shortcode (`shop-banner`) for  the banner on the shop page.

This is a list of changes:
- Added `shop-banner.php` for the new shop-banner shortcode.
- Registered the shop-banner shortcode in `functions.php`
- Added styling in these files: `shop-banner.scss` / `shop.scss` / `starburst.scss`
- Added the `wpml/wpml-woo-commerce-multilingual` to the composer.json `repositories` property and `require` list.
- Added the WooCommerce plugin via the composer install command. Source: `wpackagist-plugin/woocommerce`

### Screenshots

**Shop page**
![kk-1](https://user-images.githubusercontent.com/4222167/141322067-9cd6539d-91fd-430a-a9a9-2b6d1e4a682a.png)

**Product page**
![kk-2](https://user-images.githubusercontent.com/4222167/141322124-9000b6a4-35e2-46dd-bb5d-5b8f9bcf7869.png)

**Shop cart**
![kk-3](https://user-images.githubusercontent.com/4222167/141322144-e50c3583-2374-4c05-a56a-75b15cb4f653.png)

franei commented:
Hey @tomootes, I haven't had time to test much but I briefly looked at the version on the dev instance. A few things you're probably aware of as well:
- It seems like translation is not working yet for the wc components like "Add to cart" or "Proceed to checkout"
- Some layout issues, especially with mobile
- The "default sorting" dropdown seems redundant
- There is a link to a privacy policy that looks like some kind of boilerplate text. We have a privacy policy for the website, but the shop isn't covered in it yet of course. https://extinctionrebellion.nl/privacy


I've left a few minor comments in the code, and here are two more:

- we can't use the extinction symbol on pages with commercial content, but can call get_header() with an argument to replace the symbol in the navbar:
```php
  get_header(null, array(
    'navbar-logo' => 'xrnl-stacked-logo.svg'
  ));
```
like on this page: https://extinctionrebellion.nl/donaties/

- we have autoprefixer installed, therefore vendor prefixes aren't necessary in the scss.

tomootes commented:
@SamLubbers in succession to yesterdays call i fixed the following things:
- The header.php is now changed i a way that it changes the logo if it is on the shop page. Sadly i couldnt make it work more elegant :/ The shop page doesnt let me change it to a custom page template. Its a solid solution, but ugly.
- I added a custom page template for the other shop pages. This way we can change the logo on the cart / checkout as well.
- I added the "add to cart" link to the "In mand" button.

Other things that we discussed yesterday:
- Hiding the product page. Can be done in the wp-admin.
- Change the privacy statement. 

Thanks a lot :deciduous_tree:

SamLubbers commented on web/app/themes/xrnl/assets/sass/_variables.scss:
```diff
@@ -13,10 +13,15 @@ $xr-warm-yellow: #ffc11e;
 $xr-angry: #c80082;
 $xr-primary: $xr-green;
+
+$xr-hard-pink: #ff4677;
```
We prefix colours with `xr` only if they are in the XR style guide. Since this one isn't I think we can just call it `hard-pink`.

SamLubbers commented on web/app/themes/xrnl/assets/sass/_variables.scss:
```diff
@@ -57,3 +62,13 @@ $link-color: #000 !default;
+$font-xr-size-md: 1.25rem;
+
+/* BREAKPOINTS */
+$breakpoint-sm:	576px;
+$breakpoint-md: 768px;
+$breakpoint-lg: 992px;
+$breakpoint-xl: 1200px;
+$breakpoint-xxl: 1400px;
```
Since we use bootstrap I think we don't need to redefine breakpoints. Check out in their documentation how you can use the bootstrap breakpoints in the scss code: https://getbootstrap.com/docs/4.0/layout/overview/#responsive-breakpoints

tomootes replied:
Check! Removed them

SamLubbers commented on web/app/themes/xrnl/functions.php:
```diff
@@ -63,6 +63,7 @@ function xrnl_lg_scripts()
   register_nav_menu('primary-mobile', 'Main menu mobile');
   register_nav_menu('footer-1', 'Footer menu 1');
   register_nav_menu('footer-2', 'Footer menu 2');
+  register_nav_menu('footer-3', 'Footer menu 3');
```
is this necessary in this PR? :thinking: or does this belong to #167 (which I will review soon! :pray:)

SamLubbers commented on web/app/themes/xrnl/template-parts/shop-banner.php:
```diff
@@ -0,0 +1,50 @@
+                $post_id = get_the_ID();
+                $url = get_the_post_thumbnail_url( $post_id, 'thumbnail' );
+            ?>
+            <img src="<?php echo($url) ?>" alt="">
```
the image should have an `alt` text, no?

SamLubbers commented on web/app/themes/xrnl/template-parts/shop-banner.php:
```diff
@@ -0,0 +1,50 @@
+            <div class="price-tag">
+                <span class="price-tag__price">
+                    <div class="starburst">
+                        <span>
+                            <span>
+                                <span>
+                                    <span>
+                                        <span>
+                                            <span>
+                                                <span>
+                                                    €20,25
+                                                </span>
+                                            </span>
+                                        </span>
+                                    </span>
+                                </span>
+                            </span>
+                        </span>
+                    </div>
+                </span>
```
2 things:

1. why so many spans? :exploding_head: 
2. the price shouldn't be hardcoded in the code. Can you get that from the woo commerce product info? If not, you'll have to make a new custom field

tomootes replied:
The spans are there because it creates the shape of the start :) The price is now dynamic and comes from the product price on Woo-Commerce.

SamLubbers commented on web/app/themes/xrnl/template-parts/shop-banner.php:
```diff
@@ -0,0 +1,50 @@
+        <?php $product_obj = get_page_by_path( 'hete-klimaat-kalender', OBJECT, 'product' ); ?>
+        <div  class="d-flex justify-content-end align-items-end mt-2">
+            <a href="/?add-to-cart=<?php echo($product_obj->ID); ?>" class="in-mand">
+                IN <strong>MAND</strong>
```
This string is not translatable at the moment. [Here](https://github.com/xrnl/extinction-rebellion-nl/blob/master/CONTRIBUTING.md#beware-of-translations) you can read how you can make strings translatable.

tomootes replied:
Check! The string is translatable now.

SamLubbers commented on web/app/themes/xrnl/template-parts/shop-banner.php:
```diff
@@ -0,0 +1,50 @@
+    </div>
+</div>
+<style>
+        @media only screen and (min-width: 600px) {
+            .entry-title{
+                font-size: 4rem;
+            }
+        }
```
(I think) we always keep css in `.scss` files to have the code more organised. Could you move this to the `shop.scss` file?

tomootes replied:
Check! Removed it.

SamLubbers reviewed (changes requested):
Hey @tomootes, awesome that you've implemented the changes so quickly :raised_hands: I've deployed your branch to the [development server](https://development.extinctionrebellion.nl/heteklimaatkalender/) so I could easily test the code with the content you've already added there. Here is what I found


#### What new things are working as expected

1. dutch translations in the woo commerce plugin :partying_face: 
2. no more list of products at the bottom of `/heteklimaatkalender/`

#### What isn't working as expected

- [x] the image of the calendar is really small. It doesn't fill the entire width of the container like before. 

![image](https://user-images.githubusercontent.com/15846595/142398430-0633441b-0608-45e6-9ac4-792a65f5abc1.png)

- [ ] when I click on the `in mand` button, it goes to the home page. What I expect is that it goes to the `/cart` page, with 1 calendar added to the basket
- [ ] no payment options on the checkout page

![image](https://user-images.githubusercontent.com/15846595/142398559-799b70a5-195d-4e7a-8d60-031cfff2c1bb.png)


- [ ] the correct navigation bar with the logo: the `/cart` and `/checkout` page still have the default navigation bar. Curiously, the english version of `/heteklimaatkalender` also doesn't have the right navigation bar.
- [ ] related to the previous point: I get this error when running the application locally: `Fatal error: Uncaught Error: Call to undefined function is_shop() in /var/www/html/web/app/themes/xrnl/header.php:3` why does this happen? And do you think you could find a way to get this to work without the `is_shop()` function from woo commerce?

#### What I think is missing 

- [ ] hiding of the individual product page. I thought we would hide [this page](https://development.extinctionrebellion.nl/product/de-xr-hete-klimaat-kalender/) as it is duplicate from [the main page](https://development.extinctionrebellion.nl/heteklimaatkalender/)
- [ ] I think that `in mand` button doesn't make it clear that it can be used to buy the calendar. I would change the text to something like `koop de hete klimaatkalender`, and I would use the same style as other buttons on the website.
- [ ] another button at the bottom of the page `/heteklimaatkalender` so that people can easily buy the calendar after reading the text
- [ ] I don't see any code that links the `privacy beleid` button on the `/checkout` page, to a custom page where we have the privacy policy specific to woo commerce. Is this something that needs to be done once the website is deployed?

I also left a few comments in the code :eyes: 

I hope it doesn't take you much time to figure these things out. If you need help let me know :pray: good luck!

SamLubbers reviewed (approved):
Everything I mentioned is now fixed and working while testing locally :ok_man:
